sympy/matrices/matrixbase.py

```diff
@@ -5400,14 +5400,14 @@ def table(self, printer, rowstart='[', rowend=']', rowsep='\n',
         if S.Zero in self.shape:
             return '[]'
         # Build table of string representations of the elements
-        res = []
+        table: list[list[str]] = []
```

Comment:
Alternatively I can type-ignore this or change it in a different PR.
`res` was `list[list[str]]` and slowly transformed into a  `list[str]`. So the type would be `list[str | list[str]]` (or `Sequence[str | list[str]]` to avoid variance issues).
But then you'll get type errors at `res[-1].append(s)` (line 5410) since `str` has no `append` method.